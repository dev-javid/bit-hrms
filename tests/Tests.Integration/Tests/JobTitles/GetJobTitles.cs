namespace Tests.Integration.Tests.JobTitles
{
    public class GetJobTitles(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/job-titles?page=1&limit=10";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/JobTitles/GetJobTitles.sql");
            using HttpResponseMessage response = await Client.GetAsync(Route);

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
