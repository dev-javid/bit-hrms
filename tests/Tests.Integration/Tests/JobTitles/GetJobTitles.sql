INSERT INTO public.job_titles
(id, company_id, department_id, "name", created_by, created_on, modified_by, modified_on)
VALUES
(4001, 999, 3000, 'Test-Department-2', 0, '2024-05-03 03:31:32.227954-07', NULL, NULL),
(4002, 999, 3000, 'Test-Department-3', 0, '2024-05-03 03:31:32.227954-07', NULL, NULL),
(4003, 999, 3000, 'Test-Department-4', 0, '2024-05-03 03:31:32.227954-07', NULL, NULL),
(4004, 999, 3000, 'Test-Department-5', 0, '2024-05-03 03:31:32.227954-07', NULL, NULL);