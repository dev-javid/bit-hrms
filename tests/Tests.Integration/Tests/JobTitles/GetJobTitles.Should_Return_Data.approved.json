{
    "page": 1,
    "limit": 10,
    "total": 5,
    "items": [
        {
            "jobTitleId": 4000,
            "name": "Test-Department",
            "departmentId": 3000,
            "departmentName": "Test-Department"
        },
        {
            "jobTitleId": 4001,
            "name": "Test-Department-2",
            "departmentId": 3000,
            "departmentName": "Test-Department"
        },
        {
            "jobTitleId": 4002,
            "name": "Test-Department-3",
            "departmentId": 3000,
            "departmentName": "Test-Department"
        },
        {
            "jobTitleId": 4003,
            "name": "Test-Department-4",
            "departmentId": 3000,
            "departmentName": "Test-Department"
        },
        {
            "jobTitleId": 4004,
            "name": "Test-Department-5",
            "departmentId": 3000,
            "departmentName": "Test-Department"
        }
    ]
}